modules/programs/hexchat.nix

```
@@ -0,0 +1,267 @@
+          };
+        };
+      };
+    in mkOption {
+      type = nullOr (attrsOf modChannelOption);
+      description = "Configures '~/.config/hexchat/servlist.conf'";
+      example = literalExample ''
+        {
+          freenode = {
+            autojoin = [
+              "#home-manager"
+              "##linux"
+              "#nixos"
+            ];
+            charset = "UTF-8 (Unicode)";
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              useSSLForAllServers = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+    settings = mkOption {
+      description = "Configures '~/.config/hexchat/hexchat.conf'";
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };'';
+      type = nullOr (attrsOf str);
+    };
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+    overwriteConfigFiles = mkOption {
+      type = nullOr bool;
+      description = ''
+        Enables overwritting HexChat configuration files (hexchat.conf, servlist.conf). Any existing HexChat configuration will be lost. Certify to back-up any previous configuration before enabling this.
+        Enabling this setting is recommended, because everytime HexChat application is closed it overwrites Nix/Home-Manager provided configuration files, causing:
+        1. Nix/HM provided configuration to be out of sync with actual active HexChat configuration.
+        2. Blocking Nix/HM updates until configuration files are manually removed.'';
+      default = false;
+    };
+    theme = mkOption {
+      description =
+        "Theme package for HexChat. Expects a derivation containing decompressed theme files. '.hct' file format requires unzip decompression, as seen in example.";
+      example = ''
+        pkgs.stdenv.mkDerivation {
+          name = "hexchat-theme-MatriY";
+          buildInputs = [ pkgs.unzip ];
+          src = ./.;
+          installPhase = let
+            themeDrv = pkgs.fetchurl {
+              url = "https://dl.hexchat.net/themes/MatriY.hct";
+              sha256 = "sha256-ffkFJvySfl0Hwja3y7XCiNJceUrGvlEoEm97eYNMTZc=";
+            };
+          in "unzip ''${themeDrv} -d $out";
+        };'';
```

```suggestion
```suggestion
        stdenv.mkDerivation rec {
          name = "hexchat-theme-MatriY";
          buildInputs = [ pkgs.unzip ];
          src = fetchurl {
              url = "https://dl.hexchat.net/themes/MatriY.hct";
              sha256 = "sha256-ffkFJvySfl0Hwja3y7XCiNJceUrGvlEoEm97eYNMTZc=";
          };
          unpackPhase = "unzip ${src}";
          installPhase = "cp -r . $out";
        };
```

---

Another option would be to make this theme receive a submodule, that could be something like this:

```
{
  url = "https://dl.hexchat.net/themes/MatriY.hct";
  sha256 = "sha256-ffkFJvySfl0Hwja3y7XCiNJceUrGvlEoEm97eYNMTZc=";
}
```

And the module itself would create a package and set it in config. But the way it currently is is more flexible, so I am fine either way.